src/apis/EditUserTextEndpoint.js

```diff
@@ -0,0 +1,45 @@
+import apiRequest from '../apiBase/apiRequest';
+import NoModelError from '../apiBase/errors/NoModelError';
+import ValidationError from '../apiBase/errors/ValidationError';
+
+import PostModel from '../models2/PostModel';
+import CommentModel from '../models2/CommentModel';
+
+const TYPE_MAP = {
+  t1: CommentModel,
+  t3: PostModel,
+};
+
+const ENDPOINT = '/api/editusertext';
+
+export default {
+  post(apiOptions, data={}) {
+    const { thingId, text } = data;
+    if (!thingId || !text) {
+      throw new NoModelError(ENDPOINT);
+    }
+
+    const options = {
+      type: 'form',
+      query: {
+        raw_json: 1, // make sure html back from the server is un-escaped
+      },
+      body: {
+        api_type: 'json',
+        text,
+        thing_id: thingId,
+      },
+    };
+
+    return apiRequest(apiOptions, 'POST', ENDPOINT, options)
+      .then(apiResponse => {
+        const { body: { json }} = apiResponse.response;
+        if (json.errors.length) {
+          throw json.errors[0];
```

Comment:
I'm not sure. Private throws the json errors array. I could use the `ValidationError` class, but I don't have a strong preference, do you have a preference?